sympy/vector/parametricregion.py

```diff
@@ -0,0 +1,51 @@
+from sympy.core.basic import Basic
+
+class ParametricRegion(Basic):
+    """
+    Represents a parametric region in space.
+
+    Examples
+    ========
+
+    >>> from sympy import symbols, cos, sin, pi
+    >>> from sympy.vector import ParametricRegion
+    >>> r, theta = symbols("r theta")
+    >>> circle = ParametricRegion(r*cos(theta), r*sin(theta), (theta, 0, 2*pi))
+    """
+    def __new__(cls, *args, system=None):
```

Comment:
What about defining some precise signature? `*args` is a bit ambiguous.

Reply:
Yes, `*arg` is ambiguous. I used so we can define base scalars and bounds without nesting them in a tuple.

Reply:
what about `def __new__(cls, parameters_or_coordsys, mapping_tuple, **kwargs):` ?

Reply:
```python
ParametricRegion((r, theta), (r*cos(theta), r*sin(theta)), r=(0,1), theta=(0, pi))
ParametricRegion(C, (C.y, 3, -3), C.y=(4,6))
```
I don't think the User should pass a tuple of parameters or CoordSys3d object. 
We can easily determine the parameters using key values of the dictionary.

But we should also add an optional parameter for specifying the base scalars(see discussion above on `CoordSys3D` object) if a user does not want to user base scalars of the scalar/vector field. You mentioned using some lambda convention. Please explain that part.

What about this API @Upabjojr ?
```python
def __new__(cls, system=None, mapping_tuple, **kwargs): 
```
Example
```python
ParametricRegion(((r*cos(theta), r*sin(theta)), r=(0,1), theta=(0, pi))
ParametricRegion(system=C, (C.y, 3, -3), C.y=(4,6)) 
```

Reply:
`r=(0, 1), theta=(0, pi)` is not very intuitive, what about a dict of bounds?

What about this:
```python
ParametricRegion((r, theta), (r*cos(theta), r*sin(theta)), limits={r: (0,1), theta: (0, pi)})
ParametricRegion(C, (C.y, 3, -3), limits={C.y: (4,6)}) 
```

> You mentioned using some lambda convention. Please explain that part.

Define variable tuple, then mapping tuple.

Reply:
I am still not sure why a user needs to pass a tuple specifying parameters. SymPy can determine them using the dict of bounds. Will it be more intuitive for users?
```python
ParametricRegion((r, theta), (r*cos(theta), r*sin(theta)), limits={r: (0,1), theta: (0, pi)})
```

Reply:
Well, a dict does not have an order, while a tuple does. Plus, it's the same definition of lambdas:
```python
lambda r, theta: (r*cos(theta), r*sin(theta))
```

Reply:
+1 for passing parameter_tuple then.

Reply:
@Upabjojr What are your thoughts on the system parameter?
As discussed above, We have two options:
* Assume that the mapping of param region is to base vectors of the scalar/vector field.
```python
p = ParametricRegion((r, theta), (r*cos(theta), r*sin(theta)), limits={r: (0,1), theta: (0, pi)})
Integral(C.x**2*C.y, p)
# C.x = r*cos(theta),  C.y = r*sin(theta), C.z = C.z
Integral(R.x*R.y*R.z, p)
# R.x = r*cos(theta),  R.y = r*sin(theta), R.z = R.z
```
* The other option is to allow users if they wish to specify the coordinate system.
```python
p = ParametricRegion((r, theta), (r*cos(theta), r*sin(theta)), limits={r: (0,1), theta: (0, pi)}, system = C)
Integral(R.x*R.y*R.z, p)
# C.x = r*cos(theta),  C.y = r*sin(theta), C.z = C.z
```